client/GameComponents/Card.jsx

```diff
@@ -32,7 +32,7 @@ class Card extends React.Component {
     }
 
     isAllowedMenuSource() {
-        return this.props.source === 'play area' || this.props.source === 'agenda' || this.props.source === 'revealed plots';
+        return this.props.source === 'play area' || this.props.source === 'agenda' || this.props.source === 'revealed plots' || _.any(this.props.card.menu, action => action.canPlayFrom === this.props.source) ;
```

Comment:
Wouldn't this condition allow the hand-based menus to appear even while they weren't in hand? Example: Dolorous Edd is in hand, has the ambushing action menu. You use his ability, he's put into play, he's gets a location/source of 'play area'. This conditional would allow the menu to still appear even though he's no longer in hand.